source/daplink/interface/target_flash.c

```diff
@@ -118,11 +118,16 @@ static error_t target_flash_program_page(uint32_t addr, const uint8_t *buf, uint
     return ERROR_SUCCESS;
 }
 
-static error_t target_flash_erase_sector(uint32_t sector)
+static error_t target_flash_erase_sector(uint32_t addr)
 {
     const program_target_t *const flash = target_device.flash_algo;
 
-    if (0 == swd_flash_syscall_exec(&flash->sys_call_s, flash->erase_sector, sector * target_device.sector_size, 0, 0, 0)) {
+    // Check to make sure the address is on a sector boundary
```

Comment:
It might be good to assert here as well since this is more likely a logic error rather than a runtime error.